src/io/flutter/run/FlutterReloadManager.java

```diff
@@ -190,14 +190,16 @@ private void handleSaveAllNotification(@Nullable Editor editor) {
         return;
       }
 
-      if (hasErrors(app.getProject(), app.getModule(), editor.getDocument())) {
+      // If the analysis server detects any errors in the project, it will not perform a hot reload.
+      // This can cause hot reload to stop working needlessly when, eg, there is an analysis error in a test file.
+      // The reloadWithError option in settings is a workaround.
+      if (hasErrors(app.getProject(), app.getModule(), editor.getDocument()) && !mySettings.isReloadWithError()) {
```

Comment:
Is there an issue tracking a better fix (in flutter_tools)? If so, we could in-line the reference here.

Reply:
Done.